src/apps/external_executor/external_executor.cpp

```diff
@@ -150,36 +187,33 @@ namespace externalexecutor
         {
           auto* request_description =
             optional_request_description.mutable_optional();
-          auto& pending_request = pending_requests.front();
-          *request_description = pending_request.request_description;
-          // NB: Move for unique_ptr
-          active_request = std::move(pending_request);
+          auto pending_request = pending_requests.front();
+          *request_description = pending_request->request_description;
+          active_requests.emplace_hint(it, executor_id, pending_request);
           pending_requests.pop();
         }
 
         return ccf::grpc::make_success(optional_request_description);
       };
-
-      auto executor_auth_policy = std::make_shared<ExecutorAuthPolicy>();
       make_endpoint(
         "/externalexecutor.protobuf.KV/StartTx",
         HTTP_POST,
         ccf::grpc_command_adapter<
           google::protobuf::Empty,
           externalexecutor::protobuf::OptionalRequestDescription>(start),
-        ccf::no_auth_required)
+        executor_only)
+        .set_forwarding_required(ccf::endpoints::ForwardingRequired::Never)
         .install();
 
       auto end = [this](
                    ccf::endpoints::EndpointContext& ctx,
                    externalexecutor::protobuf::ResponseDescription&& payload)
         -> ccf::grpc::GrpcAdapterResponse<google::protobuf::Empty> {
-        if (!active_request.has_value())
+        const auto executor_id = get_caller_executor_id(ctx);
+        const auto it = active_requests.find(executor_id);
+        if (it == active_requests.end())
         {
-          return ccf::grpc::make_error(
-            GRPC_STATUS_FAILED_PRECONDITION,
-            "Not managing an active transaction - this should be called after "
-            "a successful call to StartTx");
+          return out_of_order_error;
```

Comment:
Replace this (line 212-216) with `find_active_request` call like below in line 305?

Reply:
I do the lookup manually here, so that we have the iterator to later call `erase`.